qcodes/instrument_drivers/rohde_schwarz/ZNB.py

```diff
@@ -1,16 +1,143 @@
 import logging
+import numpy as np
 from functools import partial
 from typing import Optional, Any, Tuple
 
 from qcodes import VisaInstrument, Instrument
 from qcodes import ChannelList, InstrumentChannel
 from qcodes.utils import validators as vals
-import numpy as np
-from qcodes.instrument.parameter import MultiParameter, ArrayParameter, ParamRawDataType
+from qcodes.instrument.parameter import MultiParameter, ArrayParameter, \
+    ParamRawDataType
+from qcodes.utils.deprecate import deprecate
+
 
 log = logging.getLogger(__name__)
 
 
+class FixedFrequencyTraceIQ(MultiParameter):
+    """
+    Sweep that returns the real (I) and imaginary (Q) parts of the VNA response.
+    Requires the use of the sweep type to be set to continuous wave mode.
+    See (https://www.rohde-schwarz.com/webhelp/ZNB_ZNBT_HTML_UserManual_en
+    /ZNB_ZNBT_HTML_UserManual_en.htm) under GUI reference -> sweep softtool
+    -> sweep type tab -> CW mode
+    """
+
+    def __init__(self, name: str, instrument: "ZNBChannel",
+                 npts: int, bandwidth: int, channel: int,
+                 check_cw_sweep_first: bool = True) -> None:
+        super().__init__(name, names=("", ""), shapes=((), ()))
+        self._instrument = instrument
+        self.set_cw_sweep(npts, bandwidth)
+        self._channel = channel
+        self._check_cw_sweep_first = check_cw_sweep_first
+        self.names = ('I', 'Q')
+        self.labels = (f'{instrument.short_name} I',
+                       f'{instrument.short_name} Q')
+        self.units = ('', '')
+        # TODO the question is if the time really corresponds to the time. The
+        # manual specifies there might be some additional overhead delay
+        self.setpoint_units = (('s',), ('s',))
+        self.setpoint_labels = (
+            ('time',),
+            ('time',)
+        )
+        self.setpoint_names = (
+            (f'{instrument.short_name}_time',),
+            (f'{instrument.short_name}_time',)
+        )
+
+    def set_cw_sweep(self, npts: int, bandwidth: int) -> None:
+        """
+        Needed to update config of the software parameter on sweep change
+        time setpoints tuple as needs to be hashable for look up.
+        Note: This is similar to the set_sweep functions of the frequency
+        sweep parameters.
+        """
+        t = tuple(np.linspace(0, npts/bandwidth, num=npts))
+        self.setpoints = ((t,), (t,))
+        self.shapes = ((npts,), (npts,))
```

Comment:
How are you supposed to set these back to the default.

Reply:
Or perhaps the defaults are just wrong and should be removed. I guess this always has a npts axis ?